BBallStatsTrack/src/com/example/bballstatstrack/models/Game.java

```diff
@@ -0,0 +1,406 @@
+    {
+        return mDate;
+    }
+
+    public long getDateMillis()
+    {
+        return mDate.getTime();
+    }
+
+    public GameLog getGameLog()
+    {
+        return mGameLog;
+    }
+
+    public int getHomePeriodFouls()
+    {
+        return mHomePeriodFouls;
+    }
+
+    public Team getHomeTeam()
+    {
+        return mHomeTeam;
+    }
+
+    public UUID getId()
+    {
+        return mID;
+    }
+
+    public int getPeriod()
+    {
+        return mPeriod;
+    }
+
+    public Team getTeamWithoutPossession()
+    {
+        Team team = getTeamWithPossession();
+        if( mHomeTeam.equals( team ) )
+        {
+            return mAwayTeam;
+        }
+        else if( mAwayTeam.equals( team ) )
+        {
+            return mHomeTeam;
+        }
+        else
+        {
+            return null;
+        }
+    }
+
+    public Team getTeamWithPossession()
+    {
+        return mHasBallPossession;
+    }
+
+    public String getTitle()
+    {
+        String home = mHomeTeam.getName();
+        int homeScore = mHomeTeam.getTotalScore();
+        String away = mAwayTeam.getName();
+        int awayScore = mAwayTeam.getTotalScore();
+        return home + " VS " + away + " [Final Score: " + homeScore + "-" + awayScore + "]";
+    }
+
+    public boolean isGameOngoing()
+    {
+        return mGameOngoing;
+    }
+
+    public boolean isPenalty( Team team )
+    {
+        if( team.equals( mAwayTeam ) )
+        {
+            return( mAwayPeriodFouls >= 5 );
+        }
+        else if( team.equals( mHomeTeam ) )
+        {
+            return( mHomePeriodFouls >= 5 );
+        }
+        return false;
+    }
+
+    public boolean isPeriodOngoing()
+    {
+        return mCurrentGameClock > 0;
+    }
+
+    public boolean isPossessionOngoing()
+    {
+        return mCurrentShotClock > 0;
+    }
+
+    public void nextPeriod()
+    {
+        mGameLog.nextPeriod();
+        mPeriod = mGameLog.getCurrentPeriod();
+        mPeriodLog = mGameLog.getCurrentPeriodLog();
+        initializeClocks();
+        resetPeriodFouls();
+    }
+
+    public void pauseGame()
+    {
+        mGameOngoing = false;
+    }
+
+    public void resetGameClock()
+    {
+        if( mPeriod < 4 )
+        {
+            mCurrentGameClock = mMaxGameClock;
+        }
+        else
+        {
+            mCurrentGameClock = mMaxOTGameClock;
+        }
+    }
+
+    public void resetMidShotClock()
+    {
+        mCurrentShotClock = mReducedMaxShotClock;
+    }
+
+    public void resetShotClock24()
+    {
+        int timeDifference;
+        if( mEventGameClock == -1 )
+        {
+            timeDifference = 0;
+        }
+        else
+        {
+            timeDifference = mEventGameClock - mCurrentGameClock;
+        }
+        mCurrentShotClock = MAX_SHOT_CLOCK - timeDifference;
+    }
+
+    public void setTeamWithPossession( Team team )
+    {
+        if( team.equals( mAwayTeam ) )
+        {
+            mHasBallPossession = mAwayTeam;
```

Comment:
Since it is impossible to set a team that is neither the mHomeTeam and mAwayTeam, this is updated.